### PR TITLE
Configure test retries for JUnit Acceptance Tests

### PR DESCRIPTION
**Pull request by git-phu**
## What
part of https://github.com/airbytehq/airbyte/issues/10958
Configure test retries for JUnit Acceptance Tests

## How
Use https://github.com/junit-pioneer/junit-pioneer to support annotating specific tests with retries. More [docs](https://junit-pioneer.org/docs/retrying-test/).

Straightforward enough to implement and seems to work as expected.

┆Issue is synchronized with this [Monday item](https://airbyte-bunch.monday.com/boards/2536787439/pulses/2537088281) by [Unito](https://www.unito.io)

### CODE REVIEW

**Review by davinchia — COMMENTED** [review]
Nice!

Two questions:
- is there a way to add this at the class level instead of individual functions?
- will this output tests that fail and were later retried? One thing I hope we can do is only apply this only to tests that display this behaviour and we don't necessarily have time to reproduce/immediately fix, and not have this be a bandaid for tests that aren't well written.

**Comment by git-phu** [comment]
Thanks for reviewing!
- I did not see a way to apply this at the class level. Based on the docs I only saw the `@RetryingTest` annotation which is supposed to be used as a replacement for `@Test`
- You can see how many attempts each test took but I don't remember if the retrying tests are listed at the end of the run. Let me try this out again and see

I agree too that we should only use retries for tests that we don't currently know how to fix it or are hard to fix and shouldn't treat retries as the normal solution for getting these tests to pass.

File: airbyte-tests/src/acceptanceTests/java/io/airbyte/test/acceptance/AcceptanceTests.java [hunk 1]
```diff
@@ -120,11 +120,11 @@
...
 import org.junit.jupiter.api.Order;
-import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.TestMethodOrder;
 import org.junit.jupiter.api.Timeout;
 import org.junit.jupiter.api.condition.DisabledIfEnvironmentVariable;
 import org.junit.jupiter.api.condition.EnabledIfEnvironmentVariable;
+import org.junitpioneer.jupiter.RetryingTest;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.slf4j.MDC;
```
Comment (davinchia) [inline airbyte-tests/src/acceptanceTests/java/io/airbyte/test/acceptance/AcceptanceTests.java:130]: Can we add to the javadocs here and explain why we are using the retrying test annotation? can we also add that this should be used sparingly? i.e. any new tests should not default to the annotation.

**Comment by davinchia** [comment]
@git-phu great, thanks for looking.

Added a reminder to update the javadocs.

File: airbyte-tests/src/acceptanceTests/java/io/airbyte/test/acceptance/AcceptanceTests.java (hunk 1 above)
Comment (git-phu) [reply airbyte-tests/src/acceptanceTests/java/io/airbyte/test/acceptance/AcceptanceTests.java:130]: sounds good, will do

**Review by davinchia — APPROVED** [review]
@git-phu thanks for updating the Javadocs!

Aye sorry, just realised one thing - we only have to add the retrying test annotation on the AcceptanceTests that are run for Kube since we don't see this for the docker builds. Those are the ones that do not have the DisabledifEnvrionmentVariable annotation. Feel free to merge after!

File: airbyte-tests/src/acceptanceTests/java/io/airbyte/test/acceptance/AcceptanceTests.java [hunk 2]
```diff
@@ -310,7 +321,7 @@ public void tearDown() throws ApiException, SQLException {
 
   }
 
-  @Test
+  @RetryingTest(3)
```
Comment (davinchia) [inline airbyte-tests/src/acceptanceTests/java/io/airbyte/test/acceptance/AcceptanceTests.java:324]: this is an example where we don't have to add the retrying test annotation.

File: airbyte-workers/src/test-integration/java/io/airbyte/workers/process/KubePodProcessIntegrationTest.java [hunk 3]
```diff
@@ -47,12 +47,24 @@
...
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.Timeout;
+import org.junitpioneer.jupiter.RetryingTest;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-// requires kube running locally to run. If using Minikube it requires MINIKUBE=true
-// Must have a timeout on this class because it tests child processes that may misbehave; otherwise
-// this can hang forever during failures.
+/**
```
Comment (davinchia) [inline airbyte-workers/src/test-integration/java/io/airbyte/workers/process/KubePodProcessIntegrationTest.java:54]: Thanks for cleaning this up!

**Review by davinchia — APPROVED** [review]
:shipit:

**Pull request merged**